lib/virt_autotest/utils.pm

```diff
@@ -607,4 +607,50 @@ sub check_port_state {
     return $port_state;
 }
 
+=head2 subscribe_extensions_and_modules
+
+  subscribe_extensions_and_modules(dst_machine => $machine, activate => 1/0, reg_exts => $exts)
+
+Any available extensions and modules listed out by SUSEConnect --list-extensions
+that do not require additional regcode can be subscribe directly by using command
+SUSEConnect -p [extension or module]. Subscription is to be performed on localhost
+by default if argument dst_machine is not given any other address. Deactivation is
+also supported if argument activate is given 0 explicitly. Multiple extensions or
+modules can be passed in as a single string separated by space to argument reg_exts
+to be subscribed one by one.
+
+=cut
+sub subscribe_extensions_and_modules {
+    my (%args) = @_;
+    $args{dst_machine} //= 'localhost';
```

Comment:
More convenient to let password of dst_machine be given?

Reply:
You should setup ssh connection somehow in advance which is not part of this subroutine. For example, you can call 'setup passwordless ssh' function. This subroutine just provides what its name suggests, namely 'do subscription'.

Reply:
okay, that's true in our VT code, but I was thinking about more widely usage of this function by other teams, which may not  have ssh setup done. Of course, up to you to decide how convenient this func is. Feel free to resolve this conversation if you decide to keep it as it is.

Reply:
ssh setup should be taken as a separate task. This function should not cover this.